core/src/api/oas-loader.ts

```diff
@@ -42,14 +46,48 @@ const selectSecurities = ({
   });
 };
 
+const parseJSON = async (filename: string) => {
+  const rawText: string = await fs.readFile(filename, "utf8");
+
+  try {
+    return JSON.parse(rawText);
+  } catch {
+    return null;
+  }
+};
+
+const isPostmanCollection = async (filename: string) => {
+  // Detect and parse postman collection into OAS or default to SwaggerParser.
+  const json = await parseJSON(filename);
+
+  return !!json?.info?._postman_id;
+};
+
+const parseAPI = async (filename: string) => {
+  const isPostman = await isPostmanCollection(filename);
+
+  if (isPostman) {
+    const oas = await postmanToOpenApi(filename, undefined, {
+      replaceVars: true,
+      outputFormat: "json",
+      operationId: "auto",
+      defaultTag: "General",
+    });
+
+    return JSON.parse(oas);
```

Comment:
Types inference can be done here, so we're not totally blocked by `postman-to-openapi`. However, there's still no way to infer `required` from the collection.